lib/datasource.js

```diff
@@ -775,8 +775,14 @@ DataSource.prototype.automigrate = function (models, cb) {
     models = undefined;
   }
 
+  cb = cb || utils.createPromiseCallback();
+
   if (!this.connector.automigrate) {
     // NOOP
+    if(cb.promise) {
+      process.nextTick(cb);
+      return cb.promise;
+    }
     return cb && process.nextTick(cb);
```

Comment:
Please simplify this code, `cb` is always defined thanks to L778.

I'd personally write this block as follows:

```js
if (!this.connector.automigrate) {
  process.nextTick(cb);
  return cb.promise;
}
```